### PR TITLE
Integrate Qianqian(Taihe) Music

### PR DESCRIPTION
**Pull request by Dumeng**

### CODE REVIEW

**Comment by listen1** [comment]
really exiting to have new music source!

**Comment by listen1** [comment]
bug report

- [ ] click hot playlist cover always go to same playlist 
- [ ] click next page in search result will result in blank page or N/A page number

File: js/provider/taihe.js [hunk 1]
```diff
@@ -0,0 +1,313 @@
+/* eslint-disable no-unused-vars */
+/* global async getParameterByName forge */
+function build_taihe() {
+  const axiosTH = axios.create({
+    baseURL: 'https://music.taihe.com/v1',
+  });
+  axiosTH.interceptors.request.use((config) => {
+    const params = {...config.params};
+    params.timestamp = 1614872801;// Math.round(Date.now() / 1000);
```
Comment (listen1) [inline js/provider/taihe.js:9]: fixed timestamp?

Comment (Dumeng) [reply js/provider/taihe.js:9]: This is for debugging. It should not be a hardcoded timestamp though the server doesn't verify it now.

File: js/provider/taihe.js [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+            result: playlists,
+          });
+        });
+      },
+    };
+  }
+
+  function th_parse_url(url) {
+    let result;
+    const match = /\/\/www.kugou.com\/songlist\/([0-9]+)/.exec(
```
Comment (RecluseWind) [inline js/provider/taihe.js:257]: This is wrong, url is kugou.

**Pull request merged**